### DISCUSSION TITLE
How to use sample_from_edges

### DISCUSSION THREAD

**Discussion by LuisaWerner** [pyg-team/pytorch_geometric#7721] [Q&A]
Hi, 

my goal is to sample batches from a ```Data``` object based on the ```edge index```. 
I have a graph with node features [n x d] (in``` data.x```)  and edge index [2 x num_edges] (in ```data.edge_index```), I want to create batches based on the edges index so that each batch keeps  the original features ```data.x```  as in the original object but a subset of the edges. 

The edges should be sampled subsequently. If the batch-size was 10, the first batch should have the first 10 batches in edge_index [2 x 10], the second batch the next 10 edges and so on. 

```Link_Neighbor_Loader``` does not do what I want to achieve since I don't need any neighbors but the subsequent sampling of edges. 
```Edge_Sampler_Input``` looks like being close to what I want to do but I don't know how to use ```sample_from_edges``` or how to overwrite it. 

Can someone help me with this problem? 
Thanks in advance! 
luisa

**Comment by wsad1** [comment]
pyg has a [RandomNodeLoader](https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html#torch_geometric.loader.RandomNodeLoader). You probably want a `RandomeEdgeLoader`. This can be built by modifying the `RanodomNodeLoader` code by
1. while calling `super().__init__` change `self.num_nodes` to `self.num_edges`
2. in `collate_fn` call `self.data.edge_subgraph`.

Let me know if you need help making these updates to `RandomNodeLoader`.
